interview-questions.md

```diff
@@ -7,44 +7,78 @@ Then, research the question to expand on your answer. Even if you feel you have
 
-4. Consider the Rails routes below. What is the name of the controller method that would be called by each route? What action would each of the controller methods perform?
+  rails generate migration interacts with the data allowing you to evolve your data base. instead of making modifications in SQL.
+
+
+
+
+4. Consider the Rails routes below. What is the name of the controller method that would be called by each route? What action would each of the controller methods perform? 
 
 action: "GET"    location: /students          
 
-action: "POST"   location: /students       
+shows a list of all students entered in the database
+
+action: "POST"   location: /students         
+
+creates a new form of data in the name of students
+
+action: "GET"    location: /students/new     
 
-action: "GET"    location: /students/new
+calls students/new and shows a list of new students allowing you to edit students list
 
-action: "GET"    location: /students/2  
+action: "GET"    location: /students/2       
 
-action: "GET"    location: /students/2/edit    
+accesses the student at the 2 index of the list of students to edit it
+
+action: "GET"    location: /students/2/edit   
+
+gets the student at the 2 index and allows you to edit that value in that index
 
 action: "PATCH"  location: /students/2      
 
-action: "DELETE" location: /students/2      
+update the student at the 2 index and actually edits then pushes the new value of the student at index 2 into place
+
+action: "DELETE" location: /students/2         
+
+destroy the information at students index of 2 so it destroys student number 2 or the old information in index 2
+
+
 
 
 
 5. As a developer, you are making an application to manage your to do list. Create 10 user stories that will help you get your application started. Read more about [user stories](https://www.atlassian.com/agile/project-management/user-stories).
+
+as a [user] i can view a page that allows the [user] to [make: a to do list]
+as a [user] i can view a page that allows the [user] to [view: their to do list]
+as a [user] i can view a page that allows the [user] to [add: objectives to their to do list]
+as a [user] i can view a page that allows the [user] to [remove: objectives from their to do list]
+as a [user] i can view a page that allows the [user] to [mark: objectives done on their to do list]
+as a [user] i can view a page that allows the [user] to [make: additional to do list]
+as a [user] i can view a page that allows the [user] to [add: time sensitive objectives to their to do lists]
+as a [user] i can view a page that allows the [user] to [remove: time sensitive objectives to their to do lists]
+as a [user] i can view a page that [alerts the [user]: to time sensitive objectives on their to do lists]
+as a [user] i can view a page that allows the [user] to [mark: time sensitive objectives on their to do lists as done]
```

Comment:
I like the approach. I see CRUD actions which is good!

Reply:
Thank you i wanted to make sure i made them the focus. I need to figure out how i want to lay these out, in a well structured focus map. So i can communicate with a team better than here. Also the exact buttons and inputs to get to those actions, it is more paraphrased than i would like.